### PR TITLE
[LoRA] pop the LoRA scale so that it doesn't get propagated to the weeds

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

The changes should be self-explanatory. But let me know in case of any explanation. 

I think we should maybe do a patch release for this.

### CODE REVIEW

File: src/diffusers/models/unets/unet_2d_condition.py [hunk 1]
```diff
@@ -1081,25 +1081,15 @@ def forward(
...
-            cross_attention_kwargs (`dict`, *optional*):
-                A kwargs dictionary that if specified is passed along to the [`AttnProcessor`].
-            added_cond_kwargs: (`dict`, *optional*):
-                A kwargs dictionary containin additional embeddings that if specified are added to the embeddings that
-                are passed along to the UNet blocks.
-            down_block_additional_residuals (`tuple` of `torch.Tensor`, *optional*):
-                additional residuals to be added to UNet long skip connections from down blocks to up blocks for
-                example from ControlNet side model(s)
-            mid_block_additional_residual (`torch.Tensor`, *optional*):
-                additional residual to be added to UNet mid block output, for example from ControlNet side model
```
Comment (sayakpaul) [inline src/diffusers/models/unets/unet_2d_condition.py:1100]: Duplicates.

File: src/diffusers/models/unets/unet_2d_condition.py [hunk 2]
```diff
@@ -1185,7 +1175,13 @@ def forward(
             cross_attention_kwargs["gligen"] = {"objs": self.position_net(**gligen_args)}
 
         # 3. down
-        lora_scale = cross_attention_kwargs.get("scale", 1.0) if cross_attention_kwargs is not None else 1.0
+        # we're popping the `scale` instead of getting it because otherwise `scale` will be propagated
+        # to the internal blocks and will raise deprecation warnings. this will be confusing for our users.
+        if cross_attention_kwargs is not None and "scale" in cross_attention_kwargs:
+            lora_scale = cross_attention_kwargs.pop("scale", 1.0)
```
Comment (BenjaminBossan) [inline src/diffusers/models/unets/unet_2d_condition.py:1181]: If `"scale" in cross_attention_kwargs`, we don't need a default for `pop`, right?

Comment (sayakpaul) [reply src/diffusers/models/unets/unet_2d_condition.py:1181]: Yeah that is correct.

**Review by yiyixuxu — APPROVED** [review]
thanks!

File: src/diffusers/models/unets/unet_2d_condition.py (hunk 2 above)
Comment (yiyixuxu) [inline src/diffusers/models/unets/unet_2d_condition.py:1181]: made change based on @BenjaminBossan 's suggestion https://github.com/huggingface/diffusers/pull/7338/files#r1528407445
```suggestion
        lora_scale = cross_attention_kwargs.pop("scale", 1.0)
```

**Comment by sayakpaul** [comment]
@yiyixuxu do you want me to do a patch release for this as this way the users won't get confused.

**Comment by yiyixuxu** [comment]
@sayakpaul sure
>do you want me to do a patch release for this as this way the users won't get confused.

**Comment by sayakpaul** [comment]
Cool, I will take care of it.

**Pull request merged**